server/sonar-server/src/main/java/org/sonar/server/platform/web/RoutesFilter.java

```diff
@@ -65,4 +69,67 @@ public void init(FilterConfig filterConfig) throws ServletException {
   public void destroy() {
     // Nothing
   }
+
+  private interface Route {
+    boolean accept(String path);
+
+    String apply(HttpServletRequest request);
+  }
+
+  /**
+   * Old scanners were using /batch/file.jar url (see SCANNERAPI-167)
+   */
+  private static class BatchRoute implements Route {
+
+    private static final String BATCH_WS = "/batch";
+
+    @Override
+    public boolean accept(String path) {
+      return path.startsWith(BATCH_WS + "/") && path.endsWith(".jar");
+    }
+
+    @Override
+    public String apply(HttpServletRequest request) {
+      String path = extractPath(request);
+      return format("%s%s/file?name=%s", request.getContextPath(), BATCH_WS, path.replace(BATCH_WS + "/", EMPTY));
+    }
+  }
+
+  /**
+   * Old scanners were using /batch_bootstrap url (see SCANNERAPI-167)
+   */
+  private static class BatchBootstrapRoute implements Route {
+
+    @Override
+    public boolean accept(String path) {
+      return "/batch_bootstrap/index".equals(path);
+    }
+
+    @Override
+    public String apply(HttpServletRequest request) {
+      return format("%s%s/index", request.getContextPath(), "/batch");
+    }
+  }
+
+  /**
+   * SONAR-7852 /api/sources?resource url is still used
+   */
+  private static class ApiSourcesRoute implements Route {
+
+    private static final String API_SOURCES_WS = "/api/sources";
+
+    @Override
+    public boolean accept(String path) {
+      return API_SOURCES_WS.equals(path);
+    }
+
+    @Override
+    public String apply(HttpServletRequest request) {
+      return format("%s%s/index?%s", request.getContextPath(), API_SOURCES_WS, request.getQueryString());
+    }
+  }
+
+  private static String extractPath(HttpServletRequest request) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this method into "BatchRoute". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3398)